### ISSUE TITLE
Actor crashed! Reel::StateError

### ISSUE THREAD

**Issue by mighe** [celluloid/reel#164]
Hi @tarcieri , we dug deep to better understand [this](https://github.com/kenichi/angelo/issues/21) issue and we found that it is a Reel problem and not an Angelo one.

To reproduce it consistently run the following server:

```ruby
require 'reel'

class MyServer < Reel::Server::HTTP

  def initialize(host = "127.0.0.1", port = 3000)
    super(host, port, &method(:on_connection))
  end

  def on_connection(connection)
    connection.each_request do |request|
      handle_request request
    end
  end

  def handle_request(request)
    resp = 'HELLO!!! ' * 10E6	# long response
    request.respond :ok, resp
  end

end

MyServer.run
```

As you can see we just respond with a very very long string. Try to connect to `localhost:3000` with your browser or the `wget` command.
When you let them download the entire body everything works correctly, but when you stop you process (quitting `wget` or clicking the X in the browser) Reel crashes with `Reel::StateError: already processing a request`.

It also happens with shorter responses, it is just harder to reproduce.

**Comment by tarcieri** [comment]
Seems related to #163

**Comment by mighe** [comment]
It seems fixed by #163 , can you release a new gem version? Thank you!

**Comment by tarcieri** [comment]
@digitalextremist I just made a Reel Core team and, umm, you're it for now :wink:

How do you feel about trying to release 0.6.0? Maybe make a 0.6.0.pre first?

**Comment by digitalextremist** [comment]
Roger that on taking over leadership on Reel, that's very comfortable for me.

I will audit where Reel stands and push toward shipping `0.6.0` possibly with a `-pre` first.

Will get back to you with more soon.

**Comment by tarcieri** [comment]
The first thing to do is probably go through the commit history and update the CHANGES.md with some high-level release notes about what's changed

**Comment by digitalextremist** [comment]
Well, looks like this is the only issue "blocking" `0.6.0` now, so once a2cec82 passes build under jRuby, I'll cut a gem @tarcieri. Probably under 72 hours unless the `Server::HTTPS` issue is really involved.

**Comment by digitalextremist** [comment]
Released in 0.6.0.pre1
